tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -4272,11 +4353,22 @@ class HLSLExternalSource : public ExternalSemaSource {
     }
 
     StringRef nameIdentifier = idInfo->getName();
+    IntrinsicDefIter cursor =
+        isVkNamespace ? FindIntrinsicByNameAndArgCount(
+                            g_VkIntrinsics, _countof(g_VkIntrinsics),
+                            StringRef(), nameIdentifier, Args.size())
+                      : FindIntrinsicByNameAndArgCount(
+                            g_Intrinsics, _countof(g_Intrinsics), StringRef(),
+                            nameIdentifier, Args.size());
+    IntrinsicDefIter end =
+        isVkNamespace
+            ? IntrinsicDefIter::CreateEnd(
+                  g_VkIntrinsics, _countof(g_VkIntrinsics),
+                  IntrinsicTableDefIter::CreateEnd(m_intrinsicTables))
+            : IntrinsicDefIter::CreateEnd(
+                  g_Intrinsics, _countof(g_Intrinsics),
+                  IntrinsicTableDefIter::CreateEnd(m_intrinsicTables));
```

Comment:
It would be slightly more convenient for some future plans if instead of having all these `?:` operators, you set the intrinsic tables, tableCount, and namespace to local variables that get used by these calls.

Reply:
Done